### PR TITLE
relax time constraint in duration-tck.scala (for Windows)

### PR DESCRIPTION
**Pull request by rkuhn**
It seems that on Windows Thread.sleep can actually return early even if it does not throw an exception.

review-by: @phaller @adriaanm

### CODE REVIEW

**Comment by adriaanm** [comment]
@jsuereth explained that it's likely due to running in a virtual machine

File: test/files/jvm/duration-tck.scala [hunk 1]
```diff
@@ -176,8 +176,8 @@ object Test extends App {
 
   Thread.sleep(1.second.toMillis)
 
-  { val l = dead.timeLeft; assert(l <= 1.second, s"$l > 1.second") }
-  { val l = dead2.timeLeft; assert(l <= 1.second, s"$l > 1.second") }
+  { val l = dead.timeLeft; assert(l <= 1100.millis, s"$l > 1.second") }
+  { val l = dead2.timeLeft; assert(l <= 1100.millis, s"$l > 1.second") }
```
Comment (phaller) [inline test/files/jvm/duration-tck.scala]: I guess we should also adjust the assertion message to be s"$l > 1100.millis"

**Comment by phaller** [comment]
Apart from my nitpick in the diff, LGTM

File: test/files/jvm/duration-tck.scala (hunk 1 above)
Comment (rkuhn) [inline test/files/jvm/duration-tck.scala]: The message still expresses the intent ;-) I'll amend and add a comment explaining how this came about.

**Comment by lrytz** [comment]
> @jsuereth explained that it's likely due to running in a virtual machine

interesting - how is that possible?

**Comment by rkuhn** [comment]
fixed the messages and added a comment

**Comment by phaller** [comment]
Great, thanks, Roland. LGTM

**Comment by JamesIry** [comment]
LGTM   (echoing @phaller in a way the kitty can pick up)

**Comment by adriaanm** [comment]
> interesting - how is that possible?

I assume it's that they skew the clock to make it look like the guest is running in real time, even though that isn't really the case (context switching between host and guest?).

**Comment by adriaanm** [comment]
Ah, Stefan has the winning explanation on scala-internals. Thread.sleep is spec'ed to allow for imprecision.

**Pull request merged**